[rachtan27] iP

rachtan27 opened the pull request:
# Duke
> I'm not a lazy person, I just get things done when I want to. - [Unknown](https://tfiglobalnews.com/2022/06/04/lazy-quotes/)

Sounds like you? Don't worry because Duke is here to help you get your life in order by remembering the things you need to do. It':
- text-based
- easy to learn 
- ~~fast~~ **_SUPERFAST_** to use

**All you need to do is:**
1. download it
2. add your tasks
3. let duke do the rest!  👍 

**Features**
- [x]  Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)  

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```ruby
public class Duke {
    public static void main(String[] args) {
        new Duke("data/duke.txt").run();
    }

Commit:
Add Gradle support

rachtan27 committed:
Completed Level1

rachtan27 committed:
Corrected missing bracket

rachtan27 committed:
Completed Level-2

rachtan27 committed:
Level3

rachtan27 committed:
completed level3

rachtan27 committed:
ui testing

rachtan27 committed:
completed Level-6

rachtan27 committed:
completed level6

rachtan27 committed:
completed Level-7

hongshenggg commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,189 @@
+import java.util.ArrayList;
 public class Duke {
-    public static void main(String[] args) {
+    public static void main(String[] args) throws DukeException {
```
The main method is quite long imo. Would it be possible to extract some code into other classes/methods?

hongshenggg commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,189 @@
+import java.util.Scanner;
+import java.util.ArrayList;
 public class Duke {
```
Good job on following the naming conventions of variables in this class

hongshenggg commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,14 @@
+public class Event extends Task{
```
Good job on following the naming conventions

hongshenggg commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,39 @@
+//addapted from partial solution provided in lecture notes
```
Misspelled adapted here.

hongshenggg commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,39 @@
+    public Task(String desc) {
+        this.desc = desc;
+        this.isDone = false;
+        this.type = null;
```
Is there a better way than initializing this variable to null?

hongshenggg reviewed (commented):
LGTM. Just some nits to fix.

rachtan27 committed:
Completed Level-8

rachtan27 committed:
Merge branch 'branch-Level-8'

rachtan27 committed:
Completed Level-8

rachtan27 committed:
Orgianise code to be more OOP

rachtan27 committed:
Reorganise into packages

rachtan27 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

rachtan27 committed:
Add JUnit tests for Todo and Event

rachtan27 committed:
Create JAR file

rachtan27 committed:
Added JavaDoc

rachtan27 committed:
Completed Level-9

rachtan27 committed:
Fixed bug of programme not terminating

rachtan27 committed:
Completed Level-10

rachtan27 committed:
added checkstyle

rachtan27 committed:
Add assertions to important points in code

There are certain points in Parser and Storage that have assumptions.
When parsing, saving or loading a task, we assume that there are only 3
different types of tasks. We also assume that the details of the event
should never be empty or null.

rachtan27 committed:
Improve code quality

The following changes has been made to improve the code quality
* extract getCommandWord function from parseCommand to adhere to SLAP
* change the codes for getTaskIndex, getDescTodo, getDescDeadline,
  getDescEvent to make them more readable and simple
* extract parsing of individual commands to make parseCommand less than
  30 lines (and to adhere to SLAP as well)
* extract switch-case in functions in storage to make sure codes are
  less than 30 lines and to adhere to SLAP
* remove showError in Ui because performs same function as showResponse

rachtan27 committed:
Merge pull request #2 from rachtan27/branch-A-Assertions

Add assertions to important points in code

rachtan27 committed:
Merge branch 'master' into branch-A-CodeQuality

rachtan27 committed:
Merge pull request #3 from rachtan27/branch-A-CodeQuality

Improve code quality

rachtan27 committed:
Completed C-Sort and fixed assertion bugs

rachtan27 committed:
Merge pull request #4 from rachtan27/branch-C-Sort

Completed C-Sort and fixed assertion bugs

rachtan27 committed:
Correct checkstyle errors

rachtan27 committed:
Upload screenshot of ui to docs

rachtan27 committed:
Catch find error when no keywords specified

rachtan27 committed:
Create new directory  and file when not found

rachtan27 committed:
Add exception for getTaskIndex and edited JavaDoc

rachtan27 committed:
uploaded squirtle pic to docs

rachtan27 committed:
edit docs readme
